subxt/src/events/events_type.rs

```diff
@@ -292,19 +300,27 @@ impl EventDetails {
 
     /// The name of the pallet from whence the Event originated.
     pub fn pallet_name(&self) -> &str {
-        self.event_metadata().pallet()
+        self.event_metadata().pallet.name()
     }
 
     /// The name of the event (ie the name of the variant that it corresponds to).
     pub fn variant_name(&self) -> &str {
-        self.event_metadata().event()
+        &self.event_metadata().variant.name
     }
 
-    /// Fetch the metadata for this event.
-    pub fn event_metadata(&self) -> &EventMetadata {
-        self.metadata
-            .event(self.pallet_index(), self.variant_index())
-            .expect("this must exist in order to have produced the EventDetails")
+    /// Fetch details from the metadata for this event.
+    pub fn event_metadata(&self) -> EventMetadataDetails {
+        let pallet = self
+            .metadata
+            .pallet_by_index(self.pallet_index())
+            .ok_or_else(|| MetadataError::PalletIndexNotFound(self.pallet_index()))
```

Comment:
I don't get why the ok_or_else is needed because you expect the line after?!

```suggestion
```

Reply:
Ah whoops, you're right, they aren't needed! I just copied the code from somewhere where they were needed and forgot to remove them :D